### ISSUE TITLE
Inserting a row with a null key column should be forbidden

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#7852]
Scylla does not allow a row with a missing key or part of the key, so naturally an `INSERT` request missing one of the key columns is rejected with an error. However, what happens when a key column isn't missing from the command - it does appear, but with a `null` value? E.g., `INSERT INTO table (p,c) VALUES ('hello', null)`

As is logical, Cassandra rejects such null-key insert operations, with InvalidRequest and a message "Invalid null value in condition for column c". But Scylla currently silently ignores this insert - it doesn't report any error, but also doesn't insert anything into the table.

Issue #3665 was a similar issue, but it was unclear what it was really about and whether it was ever fixed :-(

The new test `test_null.py`::`test_insert_null_key` demonstrates this bug. This bug also causes `cassandra_tests/validation/entities/frozen_collections_test.py`::`test_partition_key_usage_set` (and other tests in the same file) to fail because one of the many things they test is that one cannot use a null frozen collection as a key.

**Comment by nyh** [comment]
This issue was re-discovered by a user, @racevedoo, in #8962.

**Comment by avikivity** [comment]
Backported to 4.4, 4.5 (4.6 had it).

**Comment by nyh** [comment]
IMPORTANT NOTE: This fix was backported to 4.5 as @avikivity said above, but then reverted:

```
commit c8f14886dc2af2de10b994ccf1c6e48e9bc6feb1 (tag: scylla-4.5.3)
Author: Avi Kivity <avi@scylladb.com>
Date:   Thu Dec 23 13:03:13 2021 +0200

    Revert "cql3: Reject updates with NULL key values"
    
    This reverts commit 44c784cb79a698089c72f8db5907ad1906e0348b. It
    causes a regression without 6afdc6004c2871f2a941276a9780d743a974d9f7,
    and it is too complicated to backport at this time.
    
    Ref #9311.
```

So 4.5 is still vulnerable to this issue!
